src/block/block_compact.c

```diff
@@ -541,9 +540,20 @@ __compact_page_skip(
     else
         ++block->compact_pages_rewritten;
 
-    /* Estimate how much work is left. */
-    if (block->compact_pages_rewritten_expected == 0)
+    /*
+     * We must have reviewed at least some interesting number of pages for any estimates below to be
+     * worthwhile.
+     */
+    if (!block->compact_estimated && block->compact_pages_reviewed >= WT_THOUSAND) {
         __block_compact_estimate_remaining_work(session, block);
+        /* If no potential work has been found, exit compaction. */
+        if (block->compact_pages_rewritten_expected == 0) {
+            session->compact_state = WT_COMPACT_EXITING;
+            ret = WT_ERROR;
```

Comment:
Is there a better error code that we can return? WT_ERROR feels a little too generic, and it seems to imply that there was a genuine error, as opposed to not having enough work. (Would something like ECANCELED make more sense?)

Reply:
See b9988688c291c6401e3e4f5840c5d1a485493bf9 where I used `EINTR`.
I also removed the newly added state `WT_COMPACT_EXITING` in e1db377812932a80a70a2e5b890276097528febd now we have a very specific code when compaction is interrupted internally.

Reply:
And in the end... `ECANCELED`. Done in bf7f80d67dcac6661745127ca319124a6b0cc297.